### PR TITLE
Buy creator coin by staking native token (MVP)

### PR DESCRIPTION
**Pull request by shanev**
This is an incredibly naive MVP implementation of buying creator coins. A buyer delegates the native token in a 1:1 exchange for the creator coin. The creator coin denom is in the format "cc/[username]/[address]". When this is bridged to an ERC-20, the token name can just be the username. Currently the username is passed in as a parameter to `MsgBuyCreatorCoin`, but after user registration is implemented, it should get the username from that module.

For a v1 release, must at least:
* Buy on a bonding curve (or staircase function)
* Change delegation rewards to go to the creator, not the buyer (may have to fork the staking module for this)
* Handle slashing conditions (i.e: make this a real staking derivative)
* Implement sells

### CODE REVIEW

File: x/stake/client/cli/tx.go [hunk 1]
```diff
@@ -23,7 +24,7 @@ func NewStakeTxCmd() *cobra.Command {
 		Long: strings.TrimSpace(
 			fmt.Sprintf(`Stake on a post.
 Example:
-$ %s tx stake stake 1 "2" 500 --from mykey
+$ %s tx stake stake 1 "2" 500 starsvaloper1deadbeef --from mykey
```
Comment (JakeHartnell) [inline x/stake/client/cli/tx.go:27]: :joy:

Comment (shanev) [reply x/stake/client/cli/tx.go:27]: Hah, any opportunity to use a magic debug value https://en.wikipedia.org/wiki/Magic_number_(programming)#Debug_values.

File: x/stake/keeper/stake.go [hunk 2]
```diff
@@ -33,6 +33,72 @@ func (k Keeper) GetStake(ctx sdk.Context, vendorID uint32, postID curatingtypes.
...
+	ctx sdk.Context,
+	username string,
+	creator sdk.AccAddress,
+	buyer sdk.AccAddress,
+	valAddr sdk.ValAddress,
+	amount sdk.Int) error {
+
+	coin := sdk.NewCoin(fmt.Sprintf("%s/%s", username, creator.String()), amount)
+
+	stake, found, err := k.GetStake(ctx, 0, curatingtypes.PostID{}, buyer)
```
Comment (jhernandezb) [inline x/stake/keeper/stake.go:47]: so this is staking to empty post id(prob 0) and vendor 0 all the times? I guess this is what you meant about better using a forked staking module

**Review by jhernandezb — APPROVED** [review]

File: x/stake/keeper/stake.go (hunk 2 above)
Comment (shanev) [reply x/stake/keeper/stake.go:47]: Yeah, this is just for accounting to keep track of delegations for creator coins since it'll be needed for sells.

**Pull request merged**